File: compressString/CompressString.ts
const txt = 'abcdeee';
const txt2 = 'aaaabcdddbbbaaacccddd';

const Compress = (text: string): string => {
  const compressed: (string | number)[] = [];

  for (let i = 0; i < text.length; i++) {
    if (text[i] == text[i - 1]) {
      (compressed[compressed.length - 1] as number)++;
      continue;
    }
    compressed.push(text[i]);
    compressed.push(1);
  }

  return compressed.length < text.length ? compressed.join('') : text;
};

console.log(Compress(txt));
console.log(Compress(txt2));

File: compressString/CompressString.py
txt = 'abcdeee'
txt2 = 'aaaabcdddbbbaaacccddd'

def Compress (text: str) -> str:
  compressed = []
  for i in range(0, len(text)):
    if text[i] == text[i - 1]:
      compressed[len(compressed) - 1] += 1
      continue
    compressed.append(text[i])
    compressed.append(1)
  if len(compressed) > len(text): return text
  return "".join(map(str, compressed))

print(Compress(txt))
print(Compress(txt2))